quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,399 @@
+import com.microsoft.azure.kusto.data.exceptions.DataClientException;
+import com.microsoft.azure.kusto.data.exceptions.DataServiceException;
+import com.microsoft.azure.kusto.ingest.*;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionClientException;
+import com.microsoft.azure.kusto.ingest.exceptions.IngestionServiceException;
+import com.microsoft.azure.kusto.ingest.source.BlobSourceInfo;
+import com.microsoft.azure.kusto.ingest.source.FileSourceInfo;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.PrivateKey;
+import java.security.cert.X509Certificate;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static com.microsoft.azure.kusto.data.ClientRequestProperties.OPTION_SERVER_TIMEOUT;
+
+public class KustoSampleApp {
+    // TODO - Config:
+    //  If this quickstart app was downloaded from OneClick, kusto_sample_config.json should be pre-populated with your cluster's details
+    //  If this quickstart app was downloaded from GitHub, edit kusto_sample_config.json and modify the cluster URL and database fields appropriately
+    private static final String CONFIG_FILE_NAME = "kusto-sample-config.json";
+
+    // TODO - Config (Optional): Change the authentication method from User Prompt to any of the other options
+    //  Some of the auth modes require additional environment variables to be set in order to work (see usage in generateConnectionString below)
+    //  Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+    private static final String AUTHENTICATION_MODE = "UserPrompt"; // Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)
+
+    // TODO - Config (Optional): Toggle to false to execute this script "unattended"
+    private static final boolean WAIT_FOR_USER = true;
+
+    private static String databaseName;
+    private static String tableName;
+    private static String kustoUrl;
+    private static String ingestUrl;
+    private static boolean useExistingTable;
+    private static List<Map<String, String>> dataToIngest;
+    private static String tableSchema;
+    private static int step = 1;
+    private static final String BATCHING_POLICY = "{ \"MaximumBatchingTimeSpan\": \"00:00:10\", \"MaximumNumberOfItems\": 500, \"MaximumRawDataSizeMB\": 1024 }";
+    private static final int WAIT_FOR_INGEST_SECONDS = 20;
+
+    public enum ExecutionType {
+        CONTROL_COMMAND("Control Command"),
+        QUERY("Query");
+
+        private final String name;
+
+        ExecutionType(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    public static void main(String[] args) throws InterruptedException {

```

Comment:
In a future commit/PR, I will break `main` into submethods.
This both aligns with best practices, but also makes it easier to refactor this "query-ingest-query" approach into an additional "query-only" mode.